app/components/school_preorder_status_tag_component.rb

```diff
@@ -17,15 +18,28 @@ def type
       :yellow
     when 'school_ready', 'ready'
       :blue
+    when 'rb_can_order', 'school_can_order'
+      :green
     else
       :default
     end
   end
 
 private
 
+  attr_reader :viewer
+
   def status
-    school.preorder_status_or_default
+    @status ||= school.preorder_status_or_default
+  end
+
+  def text_key
+    hash = {
+      'trust' => :responsible_body,
+      'local_authority' => :responsible_body,
+    }
+
+    hash.fetch(viewer.class.to_s.underscore, :default)
   end
```

Comment:
I think this would be clearer if it were something like this:
```suggestion
  def text_key
    viewer.class.to_s.underscore.in?(%w[trust local_authority]) ? :responsible_body : :default
  end
```